Zelda/Zelda.cpp

```diff
@@ -1,1205 +0,0 @@
-#include <cstdlib>
-#include <fstream>
-
-using namespace std;
-
-/*Some COLOR Constants*/
-const int RED = 12;
-const int PURPLE = 13;
-const int GREEN = 10;
-const int AQUA = 11;
-const int YELLOW = 14;
-const int WHITE = 15;
-const int BLUE = 9;
-// Random comment to diff this file
-
-//---------------------Helper Functions------------------------
-
-
-
-void HelperFunctions::charactersCase(char* input)
-{
-	for (int i = 0; input[i]!='\0'; ++i)
-	{
-		if(input[i] >= 'a' && input[i] <= 'z')
-		{
-			input[i] = input[i] - 32;
-		}
-	}
-}
-
-
-
-void HelperFunctions::color(int k)
-{
-	HANDLE  hConsole;
-	hConsole = GetStdHandle(STD_OUTPUT_HANDLE);
-	SetConsoleTextAttribute(hConsole, k);
-}
-
-
-
-//---------------------Item------------------------
-
-
-Item::Item()
-{
-	name = nullptr;
-}
-
-
-
-Item::~Item()
-{
-	if(name)
-		delete[] name;
-}
-
-
-
-char* Item::getItemName()
-{
-	return name;
-}
-
-
-
-//---------------------Treasure------------------------
-
-
-
-Treasure::Treasure()
-{
-	worth = 0;
-}
-
-
-
-Treasure::~Treasure()
-{
-
-}
-
-
-
-Treasure::Treasure(char const* treasure_Name , int treasure_Worth)
-{
-	name = new char [strlen (treasure_Name) + 1];
-	strcpy_s(name,strlen (treasure_Name) + 1, treasure_Name);
-
-	worth = treasure_Worth;
-}
-
-
-
-int Treasure::getWorth()
-{
-	return worth;
-}
-
-
-
-//---------------------Weapon------------------------
-
-
-
-Weapon::Weapon()
-{
-
-}
-
-
-
-Weapon::~Weapon()
-{
-
-}
-
-
-
-Weapon::Weapon(char const* weapon_Name)
-{
-	name = new char [strlen (weapon_Name) + 1];
-	strcpy_s (name, strlen (weapon_Name) + 1, weapon_Name);
-}
-
-
-
-int Weapon::getWorth()
-{
-	return 0;
-}
-
-
-
-//---------------------Monster------------------------
-
-
-
-Monster::Monster()
-{
-	name = nullptr;
-	is_Alive = true;
-	killing_weapon = nullptr;
-}
-
-
-
-Monster::~Monster()
-{
-	if(name)
-		delete[] name;
-}
-
-
-
-Monster::Monster(char const* monster_Name, Item* weapon)
-{
-	name = new char [strlen(monster_Name) + 1];
-	strcpy_s (name,strlen (monster_Name) + 1, monster_Name);
-
-	is_Alive = true;
-	killing_weapon = weapon;
-}
-
-
-
-Item* Monster::getKillingWeapon()
-{
-	return killing_weapon;
-}
-
-
-
-void Monster::setKillingWeapon(Item* item)
-{
-	killing_weapon = item;
-}
-
-
-
-bool Monster::getLivingState()
-{
-	return is_Alive;
-}
-
-
-
-void Monster::setLivingState(bool currentState)
-{
-	is_Alive = currentState;
-}
-
-
-
-char* Monster::getMonsterName()
-{
-	return name;
-}
-
-
-
-//---------------------Princess------------------------
-
-
-
-Princess::Princess()
-{
-	is_Alive = true;
-}
-
-
-
-Princess::~Princess()
-{
-
-}
-
-
-
-bool Princess::getLivingState()
-{
-	return is_Alive;
-}
-
-
-
-void Princess::setLivingState(bool currentState)
-{
-	is_Alive = currentState;
-}
-
-
-
-//---------------------Room------------------------
-
-
-
-Room::Room()
-{
-	for (int i = 0; i < 4; ++i)
-	{
-		paths[i] = nullptr;
-	}
-
-	for (int i = 0; i < 5; ++i)
-	{
-		items_Present[i] = nullptr;
-	}
```

Comment:
both values would be better off deriving from static private integers rather than hardcoded for loops.